### PR TITLE
Add feature flag for `freeSolo` combobox

### PR DESCRIPTION
**Pull request by kgrhartlage**
### Description

Relates to [CUB-5327](https://labforward.atlassian.net/browse/CUB-5327). 

I'm open to suggestions for the name of this setting 🤔 `arbitraryValue`? `allowCustomOption`? `allowArbitraryValue`? 

[CUB-5327]: https://labforward.atlassian.net/browse/CUB-5327?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/workflow-step-template-schema.json [hunk 1]
```diff
@@ -3541,6 +3541,12 @@
             },
             "changeReason": {
               "$ref": "#/definitions/workflow-template/change-reason"
+            },
+            "customOptions": {
```
Comment (ignatiusreza) [inline src/workflow-step-template-schema.json:3545]: how about using existing `ui:options` or `ui:widget`?

Comment (kgrhartlage) [reply src/workflow-step-template-schema.json:3545]: My thinking was that being able to toggle the ability to enter arbitrary values is not an UI setting, it changes the functionality of the field more in the way of `changeReason`, `prepare` etc imo

Comment (ignatiusreza) [reply src/workflow-step-template-schema.json:3545]: all of those example you gave have subsequent effect and logic post input..

while this one only changed the input field..

Comment (kgrhartlage) [reply src/workflow-step-template-schema.json:3545]: Hmm, okay.

I guess this is more for the other PR, but the input still looks identical, just the `Add: "foo"` option no longer shows up. Was I supposed to change the input in any other way?

Comment (ignatiusreza) [reply src/workflow-step-template-schema.json]: No, why do we need to change it? :confused:

File: src/schemata/definitions/field.yml [hunk 2]
```diff
@@ -15,13 +15,18 @@ allOf:
...
           label:
             type: boolean
             default: true
             description: |
               A flag indicating whether a field of type `boolean` has a display label. Setting this to `false` will display either only a checkbox or switch control element, without any visible label.
+        customOptions:
+          type: boolean
+          default: false
+          description: |
+            A flag indicating whether fields of type `string` or `number` that are configured with `enum` options can be updated with a value that is not among the specified options. If set to `true`, the user can both select from the specified options, as well as enter any custom value that satisfies the field schema. Fields that are not of type `string` or `number` ignore this setting.
```
Comment (ignatiusreza) [inline src/schemata/definitions/field.yml:29]: lets do this, and flip the default to `true`
```suggestion
          strictEnum:
            type: boolean
            default: true
            description: |
              A flag indicating whether fields of type `string` or `number` that are configured with `enum` options can be updated with a value that is not among the specified options. If set to `true`, the user can both select from the specified options, as well as enter any custom value that satisfies the field schema. Fields that are not of type `string` or `number` ignore this setting.
```

File: src/validation.ts [hunk 3]
```diff
@@ -27,6 +27,10 @@ ajv.addKeyword({
   keyword: 'markdownDescription',
   schemaType: 'string',
 });
+ajv.addKeyword({
+  keyword: 'customOptions',
+  schemaType: 'boolean',
+});
```
Comment (ignatiusreza) [inline src/validation.ts:33]: not needed?
```suggestion
```

**Review by ignatiusreza — CHANGES_REQUESTED** [review]

File: src/validation.ts (hunk 3 above)
Comment (kgrhartlage) [reply src/validation.ts]: validation fails otherwise 👀

Comment (ignatiusreza) [reply src/validation.ts]: only because the property is not indented correctly..

File: src/workflow-step-template-schema.json (hunk 1 above)
Comment (kgrhartlage) [reply src/workflow-step-template-schema.json]: So we do want it to look the same regardless of this setting, but we also want this to be a UI setting?

**Review by ignatiusreza — APPROVED** [review]

**Pull request merged**